roles/telemetry_verify_metrics/tasks/verify_rabbitmq_metrics.yml

```diff
@@ -1,45 +1,140 @@
   changed_when: false
-  failed_when: result.rc >= 1
-  loop: "{{ cr_names.stdout_lines }}"
+  failed_when: pod_names.rc >= 1 or pod_names.stdout == ""
 
-- name: |
-    TEST Use openstack observabilityclient to verify rabbitmq metrics are stored in prometheus
+- name: Create rabbitmq pod list
+  ansible.builtin.set_fact:
+    rabbitmq_pod_list: "{{ rabbitmq_pod_list | default([]) + [item] }}"
+  loop: "{{ pod_names.stdout_lines }}"
+
+# TODO: Task  can be deleted in the future. The fixed scrapeconfig was targeted for FR4 BR1
+- name: Detect if we're working with fixed scrapeconfigs
   ansible.builtin.shell: |
-    {{ openstack_cmd }} metric query rabbitmq_identity_info{instance=\'{{ item }}.openstack.svc:15691\'} --disable-rbac
-  register: result
-  delay: 30
-  retries: 10
+    # We can use "regex: namespace" to detect the old scrapeconfig. This string is absent
+    # in the new one.
+    oc get scrapeconfig {{ rabbitmq_sc_list[0].name }} -oyaml | grep 'regex: namespace'
+  register: scrapeconfig_regex_grep_result
   changed_when: false
-  until: result.rc == 0 and "rabbitmq_identity_info" in result.stdout
-  loop: "{{ cr_names.stdout_lines }}"
+  failed_when: false
+
+# TODO: Block can be deleted in the future. The fix was targeted for FR4 BR1
+- name: Perform tests without fixed rabbitmq scrapeconfigs
+  when: scrapeconfig_regex_grep_result.rc == 0
+  block:
+  - name: Verify rabbitmq scrapeconfig exists and pods are running
+    ansible.builtin.include_role:
+      name: common
+    vars:
+      common_cr_list:
+        - kind: scrapeconfigs.monitoring.rhobs
+          name: telemetry-rabbitmq
+      common_pod_list: "{{ rabbitmq_pod_list }}"
+      common_pod_status_str: "Running"
+      common_pod_nspace: openstack
+
+  - name: Check rabbitmq metric endpoints
+    ansible.builtin.shell: |
+      oc rsh openstackclient curl https://{{ item }}.openstack.svc:15691/metrics
+    register: result
+    changed_when: false
+    failed_when: result.rc >= 1
+    loop: "{{ cr_names.stdout_lines }}"
+
+  - name: |
+      TEST Use openstack observabilityclient to verify rabbitmq metrics are stored in prometheus
+    ansible.builtin.shell: |
+      {{ openstack_cmd }} metric query rabbitmq_identity_info{instance=\'{{ item }}.openstack.svc:15691\'} --disable-rbac
+    register: result
+    delay: 30
+    retries: 10
+    changed_when: false
+    until: result.rc == 0 and "rabbitmq_identity_info" in result.stdout
+    loop: "{{ cr_names.stdout_lines }}"
+
+- name: Perform tests with fixed rabbitmq scrapeconfigs
+  when: scrapeconfig_regex_grep_result.rc >= 1
+  block:
+  - name: Verify rabbitmq scrapeconfigs exist and pods are running
+    ansible.builtin.include_role:
+      name: common
+    vars:
+      common_cr_list: "{{ rabbitmq_sc_list }}"
+      common_pod_list: "{{ rabbitmq_pod_list }}"
+      common_pod_status_str: "Running"
+      common_pod_nspace: openstack
+
+  - name: Retrieve rabbitmq pod IPs
+    ansible.builtin.shell: |
```

Comment:
This could be replaces with command, since it doesn't use pipes.

Can you try switching to command? There might be some other reason that it doesn't work, but IMHO, it's worth a try